pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -505,4 +507,39 @@ public void free() throws SQLException {
     fieldBytes = null;
     arrayList = null;
   }
+
+  @Override
+  public boolean equals(@Nullable Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+
+    PgArray other = (PgArray) obj;
+    if (oid != other.oid) {
+      return false;
+    }
+    if (fieldString == null) {
+      if (other.fieldString != null) {
+        return false;
+      }
+    } else if (!fieldString.equals(other.fieldString)) {
+      return false;
+    }
+    if (fieldBytes == null) {
+      if (other.fieldBytes != null) {
+        return false;
+      }
+    } else if (!java.util.Arrays.equals(fieldBytes, other.fieldBytes)) {
+      return false;
+    }
```

Comment:
here the null check for fieldBytes is unnecessary as Arrays.equals method implicitly handles null check.